controller/impl.go

```diff
@@ -2,14 +2,24 @@ package controller
 
 import (
 	"context"
-	"io"
+	"math"
 	"time"
 
 	client "github.com/aserto-dev/go-aserto"
 	api "github.com/aserto-dev/go-grpc/aserto/api/v2"
 	management "github.com/aserto-dev/go-grpc/aserto/management/v2"
 	"github.com/pkg/errors"
 	"github.com/rs/zerolog"
+	"golang.org/x/sync/errgroup"
+)
+
+type SleepResult bool
+
+const (
+	timeout                     = 1 * time.Second
+	maxBackoff                  = 600 * time.Second // max waits between retries.
+	Canceled        SleepResult = true
```

Comment:
It's just a style thing but `SleepResult` is essentially a two-value enum and, for clarity, we tend to group the type and value definitions together with the default value listed first:
```go
type SleepResult bool

const (
    DurationReached SleepResult = false
    Canceled        SleepResult = true
)

const (
    timeout     = 1 * time.Second
    maxBackoff = 10 * time.Minute
)
```